src/exabgp/bgp/message/update/nlri/mvpn/sharedjoin.py

```diff
@@ -0,0 +1,114 @@
+from exabgp.protocol.family import AFI
+from exabgp.protocol.family import SAFI
+
+from exabgp.bgp.message.update.nlri.qualifier import RouteDistinguisher
+from exabgp.bgp.message.update.nlri.mvpn.nlri import MVPN
+from exabgp.bgp.message.notification import Notify
+from exabgp.protocol.ip import IP
+from struct import pack
+
+# +-----------------------------------+
+# |      RD   (8 octets)              |
+# +-----------------------------------+
+# |    Source AS (4 octets)           |
+# +-----------------------------------+
+# | Multicast Source Length (1 octet) |
+# +-----------------------------------+
+# |   Multicast Source (variable)     |
+# +-----------------------------------+
+# |  Multicast Group Length (1 octet) |
+# +-----------------------------------+
+# |  Multicast Group   (variable)     |
+# +-----------------------------------+
+
+
+@MVPN.register
+class SharedJoin(MVPN):
+    CODE = 6
+    NAME = "C-Multicast Shared Tree Join route"
+    SHORT_NAME = "Shared-Join"
```

Comment:
This name may be used when the route is printed and may need to be lower case?

Reply:
You mean NAME, SHORT_NAME or both? I tried to follow example from other nlri code (like mup and evpn), but didn't notice a pattern regarding lower-case use.

Notice SHORT_NAME is already lower-cased here when printing by _prefix():
https://github.com/Exa-Networks/exabgp/blob/279a289ba4cd5a5e6ab2a6c98d494b9d6fb33e9c/src/exabgp/bgp/message/update/nlri/mvpn/sharedjoin.py#L51-L52
https://github.com/Exa-Networks/exabgp/blob/279a289ba4cd5a5e6ab2a6c98d494b9d6fb33e9c/src/exabgp/bgp/message/update/nlri/mvpn/nlri.py#L60-L61

This is the print output you mean right?
```bash
reactor       connected to peer-1 with outgoing-1 10.99.12.6-10.99.12.5
processes     ipv4 added to neighbor 10.99.12.5 local-ip 10.99.12.6 local-as 65000 peer-as 65000 router-id 32.32.32.32 family-allowed in-open : mvpn:shared-join::65000:99999:65000:10.99.199.1:239.251.255.228 extended-community target:192.168.94.12:5
processes     ipv4 added to neighbor 10.99.12.5 local-ip 10.99.12.6 local-as 65000 peer-as 65000 router-id 32.32.32.32 family-allowed in-open : mvpn:source-join::65000:99999:65000:10.99.12.2:239.251.255.228 extended-community target:192.168.94.12:5
processes     ipv6 added to neighbor 10.99.12.5 local-ip 10.99.12.6 local-as 65000 peer-as 65000 router-id 32.32.32.32 family-allowed in-open : mvpn:shared-join::65000:99999:65000:fd00::1:ff0e::1 extended-community target:192.168.94.12:5
processes     ipv6 added to neighbor 10.99.12.5 local-ip 10.99.12.6 local-as 65000 peer-as 65000 router-id 32.32.32.32 family-allowed in-open : mvpn:source-join::65000:99999:65000:fd12::2:ff0e::1 extended-community target:192.168.94.12:5
processes     ipv6 added to neighbor 10.99.12.5 local-ip 10.99.12.6 local-as 65000 peer-as 65000 router-id 32.32.32.32 family-allowed in-open : mvpn:sourcead::65000:99999:fd12::4:ff0e::1 extended-community target:65000:99999
processes     ipv4 added to neighbor 10.99.12.5 local-ip 10.99.12.6 local-as 65000 peer-as 65000 router-id 32.32.32.32 family-allowed in-open : mvpn:sourcead::65000:99999:10.99.12.4:239.251.255.228 extended-community target:65000:99999
```